src/cargo/ops/cargo_rustc/context/mod.rs

```diff
@@ -770,236 +781,25 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
-        }
-
-        // When not overridden, then the dependencies to run a build script are:
         //
-        // 1. Compiling the build script itself
-        // 2. For each immediate dependency of our package which has a `links`
-        //    key, the execution of that build script.
-        let not_custom_build = unit.pkg
-            .targets()
-            .iter()
-            .find(|t| !t.is_custom_build())
-            .unwrap();
-        let tmp = Unit {
-            target: not_custom_build,
-            profile: &self.profiles.dev,
-            ..*unit
-        };
-        let deps = self.dep_targets(&tmp)?;
-        Ok(deps.iter()
-            .filter_map(|unit| {
-                if !unit.target.linkable() || unit.pkg.manifest().links().is_none() {
-                    return None;
-                }
-                self.dep_build_script(unit)
-            })
-            .chain(Some(Unit {
-                profile: self.build_script_profile(unit.pkg.package_id()),
-                kind: Kind::Host, // build scripts always compiled for the host
-                ..*unit
-            }))
-            .collect())
-    }
-
-    /// Returns the dependencies necessary to document a package
-    fn doc_deps(&self, unit: &Unit<'a>) -> CargoResult<Vec<Unit<'a>>> {
-        let deps = self.resolve
-            .deps(unit.pkg.package_id())
-            .filter(|dep| {
-                unit.pkg
-                    .dependencies()
-                    .iter()
-                    .filter(|d| d.name() == dep.name())
-                    .any(|dep| match dep.kind() {
-                        DepKind::Normal => self.dep_platform_activated(dep, unit.kind),
-                        _ => false,
-                    })
-            })
-            .map(|dep| self.get_package(dep));
-
-        // To document a library, we depend on dependencies actually being
-        // built. If we're documenting *all* libraries, then we also depend on
-        // the documentation of the library being built.
-        let mut ret = Vec::new();
-        for dep in deps {
-            let dep = dep?;
-            let lib = match dep.targets().iter().find(|t| t.is_lib()) {
-                Some(lib) => lib,
-                None => continue,
-            };
-            ret.push(Unit {
-                pkg: dep,
-                target: lib,
-                profile: self.lib_or_check_profile(unit, lib),
-                kind: unit.kind.for_target(lib),
-            });
-            if self.build_config.doc_all {
-                ret.push(Unit {
-                    pkg: dep,
-                    target: lib,
-                    profile: &self.profiles.doc,
-                    kind: unit.kind.for_target(lib),
-                });
+        // Note there's a subtlety about this piece of code! The
+        // `build_script_overridden` map here is populated in
+        // `custom_build::build_map` which you need to call before inspecting
```

Comment:
This is here because `build_map` requires `dep_targets`, right?

Reply:
Yes!